LadybugTools_oM/ExecuteCommands/HeatPlotCommand.cs

```diff
@@ -0,0 +1,46 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Adapter;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Text;
+
+namespace BH.oM.LadybugTools
+{
+    [Description("Command that, when executed with the LadybugToolsAdapter, produces a heatmap from data in an epw file.\nOutput is a string of either the path to the image (if OutputLocation is not set) or the base 64 string representation of it.")]
+    public class HeatPlotCommand : ISimulation
+    {
+        [Description("The path to an EPW file.")]
+        public virtual FileSettings EPWFile { get; set; } = new FileSettings();
+
+        [Description("Key representing an hourly continuous collection in the EPW file to plot.")]
+        public virtual EPWKey EPWKey { get; set; } = EPWKey.Undefined;
+
+        [Description("A Matplotlib colour map. Corresponds to the 'cmap' parameter of plot methods. See https://matplotlib.org/stable/users/explain/colors/colormaps.html for examples of valid keys. Default of 'viridis'.")]
+        public virtual string ColourMap { get; set; } = "viridis";
```

Comment:
Didn't we discuss this to be an enum? I can't remember as it's been a while so apologies if we did and this was our result!

Reply:
This was after a lot of discussion, the enum `ColourMap` can be used, and is validated and converted properly to a format that matplotlib expects, but there are other possible colour maps that can be input, as well as modifiers. Currently if the command is used with a string that isn't in the ColourMap enum, it records a warning, but runs it through python anyway, with a message that tells the user that the colours on any plots may be incorrect/defaulted.